runtime/ops/process.rs

```diff
@@ -301,37 +301,66 @@ pub fn kill(pid: i32, signal: &str) -> Result<(), AnyError> {
   use deno_core::error::type_error;
   use std::io::Error;
   use std::io::ErrorKind::NotFound;
-  use winapi::shared::minwindef::DWORD;
   use winapi::shared::minwindef::FALSE;
   use winapi::shared::minwindef::TRUE;
   use winapi::shared::winerror::ERROR_INVALID_PARAMETER;
   use winapi::um::errhandlingapi::GetLastError;
   use winapi::um::handleapi::CloseHandle;
   use winapi::um::processthreadsapi::OpenProcess;
   use winapi::um::processthreadsapi::TerminateProcess;
+  use winapi::um::wincon::GenerateConsoleCtrlEvent;
+  use winapi::um::wincon::CTRL_BREAK_EVENT;
+  use winapi::um::wincon::CTRL_C_EVENT;
   use winapi::um::winnt::PROCESS_TERMINATE;
 
-  if !matches!(signal, "SIGKILL" | "SIGTERM") {
-    Err(type_error(format!("Invalid signal: {}", signal)))
-  } else if pid <= 0 {
-    Err(type_error("Invalid pid"))
-  } else {
-    let handle = unsafe { OpenProcess(PROCESS_TERMINATE, FALSE, pid as DWORD) };
+  if pid <= 0 {
+    return Err(type_error(
+      "Process id (pid) cannot be negative or zero on Windows.",
+    ));
+  }
+
+  if matches!(signal, "SIGINT" | "SIGBREAK") {
+    let is_generated = unsafe {
+      GenerateConsoleCtrlEvent(
+        match signal {
+          "SIGINT" => CTRL_C_EVENT,
+          "SIGBREAK" => CTRL_BREAK_EVENT,
+          _ => unreachable!(),
+        },
+        pid as u32,
```

Comment:
@GJZwiers Did you check that processes actually receive the `CTR_C_EVENT`? 

According to MSDN (which is often wildly inaccurate, but still) you can only send it to `CTR_C_EVENT` to all processes in the terminal.
From: https://docs.microsoft.com/en-us/windows/console/generateconsolectrlevent

> This signal cannot be limited to a specific process group. If dwProcessGroupId is nonzero, this function will succeed, but the CTRL+C signal will not be received by processes within the specified process group.

Reply:
I checked some of the examples on https://deno.land/manual/examples/os_signals and they are working with the current changes. A Deno child process also receives sigint/sigbreak on windows now, with e.g. something like this:

```js
Deno.addSignalListener("SIGINT", () => {
    console.log("parent process received SIGINT");
});

const child = Deno.spawnChild(Deno.execPath(), {
  stdin: "null",
  stdout: "inherit",
  stderr: "inherit",
  args: [
    "eval",
    `console.log("child process started");
    Deno.addSignalListener("SIGINT", () => {
      console.log("child process received SIGINT");
      Deno.exit(128);
    });
    let n = 0;
    setInterval(() => {
      console.log(n + 1);
      if (n === 2) {
        clearInterval();
        Deno.kill(Deno.pid, "SIGINT");
      }
      n++;
    }, 500);
    `,
  ],
});
```

Reply:
I was trying to write some tests for SIGINT and SIGBREAK in this PR and ran into this issue where sending a SIGINT to the child process also kills the parent in the terminal.

Reply:
For WinOS, new "child" processes, created from a console process using `CreateProcess()`, all share the same console (*unless* created with `CREATE_NEW_CONSOLE` or `DETACHED_PROCESS`). <kbd>CTRL+C</kbd> and <kbd>CTRL+BREAK</kbd> events are sent to all processes which share that console, so all processes in the console process group will receive the event/signal.

For situations requiring shared I/O with a "parent", such as a shim, I'll just use null handlers for signals in the parent while waiting for the "child" process to finish.

ref: [MS ~ Creation of a Console](https://docs.microsoft.com/en-us/windows/console/creation-of-a-console) @@ <https://archive.is/yFRT5>
ref: [MS ~ CTRL+C and CTRL+BREAK Signals](https://docs.microsoft.com/en-us/windows/console/ctrl-c-and-ctrl-break-signals) @@ <https://archive.is/6T140>